agents/plugins/mk_oracle

```diff
@@ -2917,7 +2916,28 @@ for sid in $SIDS; do
 
     # MK_DB_CONNECT could be changed by do_custom_sqls!
     do_custom_sqls
+
+}
+
+ORA_TMPDIR="$MK_VARDIR/tmp/oracle"
+mkdir -p $ORA_TMPDIR
+for sid in $SIDS; do
+    skip=$(skip_sid "$sid")
+    if [ "$skip" == "yes" ]; then
+        logging "[${sid}] [local]" "Skip"
+        continue
+    fi
+    logging "[${sid}] [local] starting task in background"
+    (sid_actions $sid > "$ORA_TMPDIR/$sid") &
+    if [[ $(jobs -r -p | wc -l) -ge $MAX_TASKS ]]; then
+        logging "[${sid}] [local] max parallel task running, waiting for new slot"
```

Comment:
Does the logging (there's no lock on logfile) work with parallel execution?

Reply:
@si-23 

Hi Simon, I will provide more feedback about logging soon, but until now we didn't have any problems with these changes even in big environments. 

About the tests, I checked the testing Framework and the currently used test for this plugin and don't see any meaningful way to add a test which can confirm the functionality of the new function. All Tests are based on running Functions from the plugin. The new changes don't have introduced new functions to test. And setting the new MAX_TASKS Variable doesn't interfere with the testable parts of the plugin neither.

Reply:
Hi Bastian,

you can test the default value of MAX_TASKS in test_mk_oracle_default_config.
In test_mk_oracle_load_config you can test if MAX_TASKS is set right if configured.

Reply:
Writing to a single file in parallel from different processes **will** do the wrong thing sooner or later, even if the single writes are atomic (which can't even be guaranteed with `echo`). So every logging done during the execution of `sid_actions` will potentially do the wrong thing. It doesn't matter if you've seen that bug yet or not.

There are various ways to fix this: Locking is one of them, writing to different log files and concatenating the results (single-threadedly) later is another. Then there's GNU `parallel`, but you can't assume that on the client.